source/request-as-event-emitter.js

```diff
@@ -227,7 +226,7 @@ module.exports = (options, input) => {
 		let backoff;
 
 		try {
-			backoff = options.retry.retries(++retryTries, error);
+			backoff = options.retry.retries(retryCount, error);
```

Comment:
This is a breaking change:

The client can provide a custom `retries` function and it expects to receive `1` for the `iteration` argument in the first retry. This breaks the contract, it's a SEMVER-MAJOR change. I'd prefer to keep the old behavior.

Reply:
> The client can provide a custom retries function and it expects to receive 1 for the iteration argument in the first retry.

@szmarczak We need a test for this.

Reply:
Yes, we need to preserve backwards compatibility. We can add it to https://github.com/sindresorhus/got/issues/639 if there's any benefit to it.

Reply:
> The client can provide a custom retries function and it expects to receive 1 for the iteration argument in the first retry. This breaks the contract, it's a SEMVER-MAJOR change. I'd prefer to keep the old behavior.

Actually, no. It's a bug. According to the docs, it should start with 0:

> Delays between retries counts with function 1000 * Math.pow(2, retry) + Math.random() * 100, where retry is attempt number (starts from 0).

Reply:
@sindresorhus 

> we need to preserve backwards compatibility

Compatibility with the docs or compatibility with the previous mechanism?

Reply:
@szmarczak I see it as explaining the default logic, not the actual arguments. I'm using a custom `retries` function so it's a breaking change.

The documentation should also be updated to document what arguments the `retries` function gets.

Reply:
In case we agree to save the previous mechanism, then yes.

@sindresorhus any thoughts?

Reply:
> @sindresorhus any thoughts?

Yes, like I said, backwards-compatibility must be preserved.

Reply:
Right. Then it starts from 1.

Reply:
@szmarczak But if you think it should be changed, we can do so in Got 10. Add a comment https://github.com/sindresorhus/got/issues/639 if so.

Reply:
I generally prefer 0-indexed collections, however in this specific case I believe it makes sense that `iteration === 1` in the first retry attempt. Having retry attempt as 0 is not intuitive.

Having the first retry attempt as 1 allows writing semantic code like:

```js
got('url', {
  retry: {
    // Retry up to 2 times.
    retries: retry => retry <= 2 ? getRetryDelay() : 0,
  },
});
```

@sindresorhus @szmarczak